src/components/navigation/SlideMenuItems.tsx

```diff
@@ -1,86 +1,123 @@
-      );
-      if (activeItem) {
-        activeItem.parents.forEach(parent => menuState.expandedNavItems.add(parent));
-      }
-    });
-  }, [history]);
+  const hasSubList = !!subItemList?.length;
+  const sublistOpen = !!menuState?.expandedNavItems.includes(id);
+  const isSelected: boolean = ((): boolean => {
+    if (menuState?.activeNavItem && menuState?.activeNavItem?.path === path) return true;
+    if (isDrawerOpen) return false;
+    if (menuState?.activeNavItem?.parents.includes(id)) return true;
+    return false;
+  })();
 
-  const handleClick = (): void => {
-    if (hasSubList) {
-      if (open) {
-        menuState.expandedNavItems.delete(props.id);
-        setOpen(false);
+  const handleClick = useCallback((): void => {
+    if (!isDrawerOpen) {
+      onRequestOpen();
+    } else if (hasSubList) {
+      if (sublistOpen) {
+        menuState?.collapseNavItem(id);
       } else {
-        menuState.expandedNavItems.add(props.id);
-        setOpen(true);
+        menuState?.expandNavItem(id);
       }
-    } else if (props.path) {
-      history.push(props.path);
-      props.menuClosingAction();
+    } else if (path) {
+      history.push(path);
+      onRequestClose();
     }
-  };
+  }, [
+    hasSubList,
+    path,
+    isDrawerOpen,
+    sublistOpen,
+    onRequestOpen,
+    menuState,
+    id,
+    history,
+    onRequestClose,
+  ]);
 
-  const expandButton = open ? <ExpandLess /> : <ExpandMore />;
+  const expandButton = isDrawerOpen && sublistOpen ? <ExpandLess /> : <ExpandMore />;
 
   const subList: React.ReactNode = useMemo(
     () =>
       hasSubList ? (
-        <Collapse in={open} timeout="auto">
+        <Collapse in={isDrawerOpen ? sublistOpen : false} timeout="auto">
           <List className={classes.nestedListItem} disablePadding>
-            {props.subItemList
+            {subItemList
               ?.map((item: NavigationItem) => (
-                <NavListItem key={item.id} {...item} menuClosingAction={props.menuClosingAction} />
+                <NavListItem
+                  key={item.id}
+                  {...item}
+                  onRequestOpen={onRequestOpen}
+                  onRequestClose={onRequestClose}
+                  isDrawerOpen={isDrawerOpen}
+                />
               ))
               .filter(Boolean)}
           </List>
         </Collapse>
       ) : null,
-    [classes.nestedListItem, hasSubList, open, props.menuClosingAction, props.subItemList],
+    [
+      hasSubList,
+      isDrawerOpen,
+      sublistOpen,
+      classes.nestedListItem,
+      subItemList,
+      onRequestOpen,
+      onRequestClose,
+    ],
   );
 
-  if (props.isVisible && !props.isVisible()) {
+  useEffect(() => {
+    if (isDrawerOpen && menuState?.activeNavItem?.parents.includes(id)) {
+      menuState.expandNavItem(id);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isDrawerOpen]);
+
+  if (isVisible && !isVisible()) {
```

Comment:
this isn't used anywhere, can it be removed? I feel it would be better to instead each navItem a permissions requirement, and then check if the user meets that requirement in this component later on. for now this is just bloat

Reply:
Considering how free form permissions might end up being, I'm hesitant to support an api we don't enforce. Besides, a consumer could depend on a variety of things outside of just permissions to display a nav item which is why I think a callback is the way to go. I know it's not used right now because we don't have the auth workflow fully set up but it doesn't hurt so much that it warrants being cut as bloat.

Reply:
sounds good, I hadn't considered there may factors besides permissions. it doesn't hurt to keep it